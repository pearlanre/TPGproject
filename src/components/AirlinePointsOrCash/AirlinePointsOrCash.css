@import url(https://fonts.googleapis.com/css?family=Roboto:300);
.airline-title{
    margin: 30px 15px 15px 15px;
}
