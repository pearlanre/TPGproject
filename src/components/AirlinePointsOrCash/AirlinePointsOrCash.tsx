import React from 'react';

import { ILoyalty } from '../../interfaces/ILoyalty';
import { PointsOrCash } from '../../utilities/pointsOrCash/PointsOrCash';

interface Props {
  data: ILoyalty;
}

export const AirlinePointsOrCash: React.FunctionComponent<Props> = ({
  data,
}: Props) => {
  return (
    <div>
      <p>Select An Airline Rewards Program</p>
      <PointsOrCash data={data} program="Ticket" />
    </div>
  );
};

export default AirlinePointsOrCash;
