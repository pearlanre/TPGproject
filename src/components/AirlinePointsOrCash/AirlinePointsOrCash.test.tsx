import React from 'react';
import { shallow, mount } from 'enzyme';
import { AirlinePointsOrCash } from './AirlinePointsOrCash';

describe('AirLinePointsOrCash', () => {
  const props = {
    data: {
      name: 'Private airline',
      tpg_valuation: 2,
      type: 'airline',
    },
    program: '',
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AirlinePointsOrCash {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render the PointsOrCash Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render AirlinePointToCash', () => {
    const airline = 'Select An Airline Rewards Program';
    expect(wrapper.find('p').text()).toEqual(airline);
    expect(wrapper).toHaveLength(1);
  });
});
