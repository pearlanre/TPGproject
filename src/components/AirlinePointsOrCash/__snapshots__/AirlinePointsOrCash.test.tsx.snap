// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirLinePointsOrCash should render the PointsOrCash Component correctly 1`] = `
<div>
  <p>
    Select An Airline Rewards Program
  </p>
  <PointsOrCash
    data={
      Object {
        "name": "Private airline",
        "tpg_valuation": 2,
        "type": "airline",
      }
    }
    program="Ticket"
  />
</div>
`;
